File: TestComplete/Python_tests/PythonTests/PyWebsiteTests/Script/EventControl1_OnStartTest.py
﻿def EventControl1_OnStartTest(Sender):
  
  CONST_INPUT = "Input"
  
  Log.Message("Start")

  found = False
  i = 1
  while (i <= BuiltIn.ParamCount() and not found):
    parm = BuiltIn.ParamStr(i);
    Log.message(parm)
    p = aqString.Find(parm, "Data", 0, True);

    found = p == 0;
    
    if (found):
      parts = parm.split("=");
      if not Project.Variables.VariableExists(CONST_INPUT):
          Project.Variables.AddVariable(CONST_INPUT, "String")
      Log.Message(parts[1])

      Project.Variables.VariableByName[CONST_INPUT] = parts[1]
    
    i+=1   

File: TestComplete/Webtests/Script/EventControl1_OnStartTest.js
function EventControl1_OnStartTest(Sender) {
  
  const INPUT = "Input";
  
  Log.Message("Start")

  var found = false;

  for (let i = 1; i <= BuiltIn.ParamCount() && !found; i++) {
    let parm = BuiltIn.ParamStr(i);
    
    Log.Message(parm);
    let p = aqString.Find(parm, "Data", 0, true);
    
    found = p == 0; 

    if (found) {
       var parts = parm.split("=");
    
       if (!Project.Variables.VariableExists(INPUT)) {
          Project.Variables.AddVariable(INPUT, "String");
       }

       Project.Variables.$set("VariableByName", INPUT, parts[1]);
    }
  }
}